### PR TITLE
Porting `transaction` examples from `ethers`

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Port transaction examples from `ethers` to `alloy`

## Solution

Added the following examples
- `decode_input`
- `transfer_eth`
- `transfer_erc20`
- `trace_transaction`
- `trace_call` - Uses trace_call as debug_traceCall is pending implementation
- `gas_price_usd`

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Comment by yash-atreya** [comment]
`ens` resolution can be implemented after extension crate for it has been completed.

File: examples/transactions/examples/transfer_erc20.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        to: Some(contract_address),
+        input: Some(input).into(),
+        ..Default::default()
+    };
+
+    let pending_tx = provider.send_transaction(transfer_tx).await?;
+
+    let tx_hash = pending_tx.tx_hash().to_owned();
+
+    let _tx = provider.get_transaction_by_hash(tx_hash).await?;
```
Comment (onbjerg) [inline examples/transactions/examples/transfer_erc20.rs:40]: unused?

File: examples/transactions/examples/transfer_erc20.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        input: Some(bytecode).into(),
+        to: None,
+        ..Default::default()
+    };
+
+    // Deploy the contract
+    let pending_tx = provider.send_transaction(tx_req).await?;
+
+    // Wait for the transaction to be mined
+    let _tx = provider.get_transaction_by_hash(pending_tx.tx_hash().to_owned()).await?;
```
Comment (onbjerg) [inline examples/transactions/examples/transfer_erc20.rs:76]: use pending tx to wait for the confirmation here

File: examples/transactions/examples/transfer_eth.rs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+use reqwest::Client;
+#[tokio::main]
+async fn main() -> Result<()> {
+    let (provider, _anvil) = init();
+
+    // Transfer 1ETH from 0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266 to Address::ZERO
+    let input =
+    "f865808477359400825208940000000000000000000000000000000000000000018082f4f5a00505e227c1c636c76fac55795db1a40a4d24840d81b40d2fe0cc85767f6bd202a01e91b437099a8a90234ac5af3cb7ca4fb1432e133f75f9a91678eaf5f487c74b"
+    ;
+    let bytes = hex::decode(input).unwrap();
```
Comment (onbjerg) [inline examples/transactions/examples/transfer_eth.rs:17]: formatting

**Review by onbjerg — CHANGES_REQUESTED** [review]

**Comment by yash-atreya** [comment]
Moved https://github.com/alloy-rs/examples/pull/4

**Pull request closed without merging**